### ISSUE TITLE
Animated ScrollView contentOffset yields wrong animated value on mount inconsistantly.

### ISSUE THREAD

**Issue by nhannah** [software-mansion/react-native-reanimated#2345]
## Description

When using `contentInset` and `contentOffset`, 

```
contentInset={{ top: 100 }}
contentOffset={{ y: -100, x: 0 }}
```

with useAnimatedScrollHandler,

```
  const scroll = useSharedValue(0);

  const scrollHandler = useAnimatedScrollHandler(({ contentOffset: { y } }) => {
    scroll.value = y;
  });
```

in combination with an animated style,

```
const translationY = useDerivedValue(() => {
    const startScroll = Platform.OS === 'ios' ? -100 : 0;
    const endScroll = Platform.OS === 'ios' ? 0 : 100;

    return interpolate(
      scroll.value,
      [startScroll, endScroll],
      [0, -100],
      Extrapolate.CLAMP,
    );
  }, [scroll]);

  const animatedHeaderStyle = useAnimatedStyle(
    () => ({
      transform: [
        {
          translateY: translationY.value,
        },
      ],
    }),
    [translationY],
 );
```

the animated style returns the scroll position at `0` sometimes on mount, despite the scroll updating to the correct position. Moving the scrollview corrects this and the translation is corrected.

This can be worked around by adding a hacky fix, but there still is a delay at times where the view is in the wrong (`0`) position for a second.

```
 style={[
          animatedHeaderStyle,
          {
            transform: [
              {
                translateY: translationY.value, // Fix for ScrollView offset and reanimated issue on mount
              },
            ],
          },
        ]}
```

### Expected behavior

When using `contentInset` and `contentOffset` (on Android paddingTop is used instead of contentInset for the same effect), the initial scroll position is calculated based on the contentOffset.

### Actual behavior & steps to reproduce

Create a scrollview and animated view as described and apply the provided styles. Open the ios simulator, the animated view will appear in the incorrect location inconsistently on mount.

## Snack or minimal code example

https://snack.expo.dev/@nhannah/reanimated-2-scroll-offset-issue

## Package versions

- React Native: 0.64.0
- React Native Reanimated: 2.1.0
- NodeJS: 16.8.0

## Affected platforms

- [ ] Android
- [x] iOS
- [ ] Web

**Comment by nhannah** [comment]
Just a note, starting the `sharedValue` at the offset seems to possibly fix this, but seems like a sorta janky answer as it does not allow for measuring the header content, then adjusting to that height.

**Comment by nhannah** [comment]
I have also seen similar issues with initial values in an image viewer I created: https://github.com/GetStream/stream-chat-react-native/blob/48c5ccadaf686d89a5f8a184329a0e0092d08691/package/src/components/ImageGallery/ImageGallery.tsx#L1101

You can easily run the demo of that repo and it's linked, so you can take out those values and see the initial values are incorrect without them.

**Comment by jakub-gonet** [comment]
Hey, have you tried upgrading to 2.3.x? If I remember correctly, this may be fixed in 2.3-beta.

**Comment by ChrisYohann** [comment]
Hi, I'm having this issue in `2.3.1`, which is the latest supported version by Expo SDK 44. Regardless of which `contentOffset` I specify, there is always first an event fired with `event.contentOffset = { x : 0, y: 0 }` before getting the right values (e.g `{ x : 0, y: contentOffset }`). What can I do since the #2580 fix is only in 2.5.0 ? Thanks !
